### PR TITLE
stdin multiple file fixes

### PR DESCRIPTION
**Pull request by yoniko**
Fixes for https://github.com/facebook/zstd/issues/3206 - bugs when handling stdin as part of multiple files.

### CODE REVIEW

**Comment by yoniko** [comment]
I haven't added tests for the cases where we fail on stdin/stdout being a console as our test suite actually doesn't use the console.
Here are the outputs of the manual checks:
```
❯ ./zstd ./file1 - -c
stdin is a console, aborting
❯ ./zstd - -c
stdin is a console, aborting
❯ zstd - -c
stdin is a console, aborting
❯ ./zstd ./file1 -
stdin is a console, aborting
❯ ./zstd ./file1 - -c
stdin is a console, aborting
❯ ./zstd - -c
stdin is a console, aborting
❯ ./zstd -
stdin is a console, aborting
❯ ./zstd - -f
aaaaa
(�/�X1aaaaa
B�%
❯ ./zstd - ./file1 -f
dwadawd
Compress: 1/2 files. Current: /*stdin*\ Read:     0   B  ==>  0%(�/�XAdwadawd
  2 files compressed :285.71%   (    14   B =>     40   B)
❯ echo "abc" | ./zstd -
stdout is a console, aborting
❯ echo "abc" | ./zstd - -c
(�/�X!abc
-nL�%
❯ echo "abc" | ./zstd ./file1 -
stdout is a console, aborting
❯ echo "abc" | ./zstd ./file1 - -c
(�/�$1file1
C0W(�/�X!abc
-nL�%
```

File: tests/cli-tests/compression/multiple-files.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+cat file2.zst | zstd -d
+
+echo "Test zstd -d ./file1.zst - file2.zst"
+rm ./file1 ./file2
+echo "stdin" | zstd - | zstd -d ./file1.zst - file2.zst
+cat file1
+cat file2
+
+echo "zstd -d ./file1.zst - file2.zst -c"
+echo "stdin" | zstd | zstd -d ./file1.zst - file2.zst -c
```
Comment (terrelln) [inline tests/cli-tests/compression/multiple-files.sh]: nit: newline at end of file

**Review by terrelln — APPROVED** [review]
LGTM

File: tests/cli-tests/compression/multiple-files.sh (hunk 1 above)
Comment (yoniko) [reply tests/cli-tests/compression/multiple-files.sh]: Fixed

**Pull request merged**